pkg/sql/tests/sysbench_test.go

```diff
@@ -0,0 +1,630 @@
+}
+
+func randCValue(rng *rand.Rand) cValue {
+	return cValue(randutil.RandString(rng, 119, randutil.PrintableKeyAlphabet))
+}
+
+func randPadValue(rng *rand.Rand) padValue {
+	return padValue(randutil.RandString(rng, 59, randutil.PrintableKeyAlphabet))
+}
+
+// sysbenchDriver is capable of running sysbench.
+type sysbenchDriver interface {
+	// Transaction orchestration operations.
+	Begin()
+	Commit()
+	// Read-only operations.
+	PointSelect(tableNum, rowID)
+	SimpleRange(tableNum, rowID, rowID)
+	SumRange(tableNum, rowID, rowID)
+	OrderRange(tableNum, rowID, rowID)
+	DistinctRange(tableNum, rowID, rowID)
+	// Read-write operations.
+	IndexUpdate(tableNum, rowID)
+	NonIndexUpdate(tableNum, rowID, cValue)
+	DeleteInsert(tableNum, rowID, kValue, cValue, padValue)
+
+	// Prepares the schema and connection for the workload.
+	prep(rng *rand.Rand)
+}
+
+const (
+	sysbenchDB          = `sysbench`
+	sysbenchTableFmt    = `sbtest%d`
+	sysbenchCreateDB    = `CREATE DATABASE ` + sysbenchDB
+	sysbenchCreateTable = `CREATE TABLE sbtest%d(
+	  id INT8 PRIMARY KEY,
+	  k INT8 NOT NULL DEFAULT 0,
+	  c CHAR(120) NOT NULL DEFAULT '',
+	  pad CHAR(60) NOT NULL DEFAULT ''
+	)`
+	sysbenchCreateIndex = `CREATE INDEX k_%[1]d ON sbtest%[1]d(k)`
+
+	sysbenchStmtBegin          = `BEGIN`
+	sysbenchStmtCommit         = `COMMIT`
+	sysbenchStmtPointSelect    = `SELECT c FROM sbtest%d WHERE id=$1`
+	sysbenchStmtSimpleRange    = `SELECT c FROM sbtest%d WHERE id BETWEEN $1 AND $2`
+	sysbenchStmtSumRange       = `SELECT sum(k) FROM sbtest%d WHERE id BETWEEN $1 AND $2`
+	sysbenchStmtOrderRange     = `SELECT c FROM sbtest%d WHERE id BETWEEN $1 AND $2 ORDER BY c`
+	sysbenchStmtDistinctRange  = `SELECT DISTINCT c FROM sbtest%d WHERE id BETWEEN $1 AND $2 ORDER BY c`
+	sysbenchStmtIndexUpdate    = `UPDATE sbtest%d SET k=k+1 WHERE id=$1`
+	sysbenchStmtNonIndexUpdate = `UPDATE sbtest%d SET c=$2 WHERE id=$1`
+	sysbenchStmtDelete         = `DELETE FROM sbtest%d WHERE id=$1`
+	sysbenchStmtInsert         = `INSERT INTO sbtest%d (id, k, c, pad) VALUES ($1, $2, $3, $4)`
+)
+
+// sysbenchSQL is SQL-based implementation of sysbenchDriver.
+type sysbenchSQL struct {
+	ctx  context.Context
+	conn *pgx.Conn
+	stmt struct {
+		begin          string
+		commit         string
+		pointSelect    [sysbenchTables]string
+		simpleRange    [sysbenchTables]string
+		sumRange       [sysbenchTables]string
+		orderRange     [sysbenchTables]string
+		distinctRange  [sysbenchTables]string
+		indexUpdate    [sysbenchTables]string
+		nonIndexUpdate [sysbenchTables]string
+		delete         [sysbenchTables]string
+		insert         [sysbenchTables]string
+	}
+}
+
+func newSysbenchSQL(ctx context.Context, b *testing.B) (sysbenchDriver, func()) {
+	st := cluster.MakeTestingClusterSettings()
+	// NOTE: disabling background work makes the benchmark more predictable, but
+	// also moderately less realistic.
+	disableBackgroundWork(st)
+	s := serverutils.StartServerOnly(b, base.TestServerArgs{Settings: st})
```

Comment:
https://github.com/cockroachdb/cockroach/issues/133307#issuecomment-2438199014 suggests this might not be as close to the real thing as we like. Leaving this comment as a discussion placeholder.